pages/mesosphere/dcos/services/beta-jenkins/4.0.0-2.204.2-beta/jenkins-auth/index.md

```diff
@@ -0,0 +1,278 @@
+---
+layout: layout.pug
+navigationTitle:  Provisioning Jenkins
+title: Provisioning Jenkins
+menuWeight: 15
+excerpt: >
+  This topic describes when and how to
+  provision Jenkins with a service
+  account.
+featureMaturity:
+enterprise: true
+---
+
+# About provisioning Jenkins with a service account
+
+Whether you can or must provision Jenkins with a service account varies by [security mode](/mesosphere/dcos/latest/security/ent/#security-modes).
+
+- `permissive`: optional
+- `strict`: required
+
+To increase the security of your cluster and conform to the principle of least privilege, we recommend provisioning Jenkins with a service account in `permissive` mode. Otherwise, Marathon and Metronome will act as if Jenkins was provisioned with a service account which has the `superuser` permission.
+
+To set up a service account for Jenkins, complete the following steps.
+
+1. [Create a key pair.](#create-a-keypair)
+1. [Create a service account.](#create-a-service-account)
+1. [Create a service account secret.](#create-an-sa-secret)
+1. [Provision the service account with the necessary permissions.](#give-perms)
+1. [Create a config.json file.](#create-json)
+
+**Requirement:** In `strict` mode, the name of the service account must match the name that the service uses as its `principal`. By default, Jenkins uses `jenkins-principal` as the name of its `principal`. That's the value that we use in the following procedures. Should you modify the default, you must change `jenkins-principal` throughout to match.
+
+**Note:** We will use `jenkins-secret` as the name of the secret, `jenkins-private-key.pem` as the name of the file containing the private key, and `jenkins-public-key.pem` as the name of the file containing the public key. We recommend using these names, as it will make it easier to copy and paste the commands. If you change the names, make sure to modify the commands before issuing them.
+
+**Important:** We store the secret in the `jenkins` path. This protects it from other services, so we do not recommend changing this.
```

Comment:
Did you verify that it works? I'm not sure whether the secret needs to be stored under the service id `jenkins` or the principal name `jenkins-principal`.

Reply:
Yes, all these commands have been tried out and tested.